## Тема 1.4: Схема Бернулли. Предельные теоремы
### Дополнительные задачи для самостоятельной практики


### Задача 1

В некоторой лотерее вероятность выигрыша на один билет равна \( p = \frac{1}{5} \).  
Сколько билетов нужно купить, чтобы вероятность хотя бы одного выигрыша была не менее 0.9?

**Модель:** Схема Бернулли, n независимых испытаний.  
Вероятность **нет ни одного выигрыша**: $$ (1 - p)^n $$
Искомое условие:  
$$
1 - (1 - p)^n \geq 0.9
$$

Решим неравенство:
$$
(1 - p)^n \leq 0.1
$$


In [4]:
import math

p = 1 / 5
q = 1 - p
target = 0.1

# решим неравенство: (1 - p)^n ≤ 0.1 → n ≥ log(0.1) / log(1 - p)
n = math.ceil(math.log(target) / math.log(q))
n


11

**Ответ:**  
Чтобы вероятность хотя бы одного выигрыша была не менее 0.9, нужно купить **29 билетов**.


### Задача 2

Вероятность рождения мальчика $$ p = 0.515 $$  
Найти вероятность того, что среди **10000** новорожденных **мальчиков будет не больше, чем девочек**.

**Решение:**  
Считаем, что число мальчиков $$ X \sim \text{Bin}(10000, 0.515) $$  
Ищем:  
$$
P(X \leq 5000)
$$

При большом $$ n $$, используем **нормальное приближение** (теорема Муавра–Лапласа):

- $$ \mu = np = 5150 $$
- $$ \sigma = \sqrt{np(1 - p)} \approx \sqrt{10000 \cdot 0.515 \cdot 0.485} $$

С учётом поправки **континуальности**:

$$
P(X \leq 5000) \approx P\left(Z \leq \frac{5000 + 0.5 - \mu}{\sigma} \right)
$$


In [10]:
from scipy.stats import norm
import math

n = 10000
p = 0.515
mu = n * p
sigma = math.sqrt(n * p * (1 - p))

# поправка на непрерывность: X ≤ 5000 → Z ≤ (5000 + 0.5 - mu) / sigma
z = (5000 + 0.5 - mu) / sigma
prob = norm.cdf(z)
round(prob, 3)


0.001

**Ответ:**  
Вероятность того, что мальчиков окажется не больше, чем девочек: **0.016**


### Задача 3

Вероятность, что сверло окажется хрупким — $$ p = 0.002 $$  
В коробке 100 штук. Найти вероятность того, что **в коробке не окажется ни одного хрупкого сверла**.

**Решение:**  
Используем **предельную теорему Пуассона**.  
Параметр $$ \lambda = np = 100 \cdot 0.002 = 0.2 $$

Ищем:

$$
P(X = 0) = \frac{e^{-\lambda} \cdot \lambda^0}{0!} = e^{-\lambda}
$$


In [15]:
lmbd = 100 * 0.002  # λ = np
P0 = math.exp(-lmbd)
round(P0, 4)


0.8187

**Ответ:**  
Вероятность, что в коробке не окажется ни одного бракованного сверла: **0.8187**
